api/v1beta1/openstackprovisionserver_types.go

```diff
@@ -69,6 +69,10 @@ type OpenStackProvisionServerSpec struct {
 	OSImageDir *string `json:"osImageDir"`
 	// OSContainerImageURL - Container image URL for init with the OS qcow2 image (osImage)
 	OSContainerImageURL string `json:"osContainerImageUrl"`
+	// +kubebuilder:validation:Optional
```

Comment:
I think the provision server only needs to be created when osImageDeploymentType is SelfExtracting, so no changes need to be made to this spec.